src/EcomDestinationWriter.cs

```diff
@@ -1691,9 +1714,77 @@ private void WritePrices(Dictionary<string, object> row, Dictionary<string, Colu
             LastPriceId = LastPriceId + 1;
             dataRow["PriceID"] = "ImportedPRICE" + LastPriceId;
         }
-        if (!columnMappings.ContainsKey("PriceCurrency"))
+        if (!columnMappings.TryGetValue("PriceCurrency", out var priceCurrencyColumn))
+        {
+            dataRow["PriceCurrency"] = Ecommerce.Services.Currencies.GetDefaultCurrency().Code;
+        }
+        else
         {
-            dataRow["PriceCurrency"] = String.Empty;
+            var priceCurrencyValue = (string)row[priceCurrencyColumn.SourceColumn.Name];
+            if (string.IsNullOrWhiteSpace(priceCurrencyValue))
+            {
+                row[priceCurrencyColumn.SourceColumn.Name] = Ecommerce.Services.Currencies.GetDefaultCurrency().Code;
+            }
+        }
+
+        if (columnMappings.TryGetValue("PriceUserId", out var priceAccessUserColumn))
+        {
+            var userIdLookupValue = GetMergedValue(priceAccessUserColumn, row);
+            var userIDs = ExistingUsers.Select("AccessUserID='" + userIdLookupValue + "'").Select(r => r["AccessUserID"].ToString()).ToList();
+            if (userIDs.Count == 0)
+            {
+                userIDs = ExistingUsers.Select("AccessUserExternalID='" + userIdLookupValue + "'").Select(r => r["AccessUserID"].ToString()).ToList();
+            }
+
+            if (userIDs.Count == 1)
```

Comment:
maybe >=1? What is there are several users with the same external id? users data can always be un-predictable :)

Reply:
true, but in this case i rather have nothing/the source-value on the mapping (bad-mapped-job) than just take the first in a result of 3 :) .. but maybe @frederik5480 have an opinion on this type of logic? :)